src/mappings/gns.ts

```diff
@@ -80,25 +81,11 @@ export function handleSubgraphMetadataUpdated(event: SubgraphMetadataUpdated): v
 
   let hexHash = addQm(event.params.subgraphMetadata) as Bytes
   let base58Hash = hexHash.toBase58()
-  let metadata = ipfs.cat(base58Hash)
+
   subgraph.metadataHash = event.params.subgraphMetadata
-  if (metadata !== null) {
-    let tryData = json.try_fromBytes(metadata as Bytes)
-    if (tryData.isOk) {
-      let data = tryData.value.toObject()
-      subgraph.description = jsonToString(data.get('description'))
-      subgraph.image = jsonToString(data.get('image'))
-      subgraph.displayName = jsonToString(data.get('displayName'))
-      subgraph.codeRepository = jsonToString(data.get('codeRepository'))
-      subgraph.website = jsonToString(data.get('website'))
-    } else {
-      subgraph.description = ''
-      subgraph.image = ''
-      subgraph.displayName = ''
-      subgraph.codeRepository = ''
-      subgraph.website = ''
-    }
-  }
+
+  fetchSubgraphMetadata(subgraph, base58Hash)
```

Comment:
@davekaj Are we 100% sure that passing the entity and updating it in another function (without returning it) works as expected?

We usually did this and returned the resulting entity so that we could replace the variable pointer afterwards to make sure the `.save()` function actually saved the changes we did on the auxiliary function, but I can't remember if that was because of some bug or because of how reference/pointers worked in `AssemblyScript` 🤔 

Also, if I recall correctly, the build stage usually whines if we pass an entity type parameter without explicitly casting it (`subgraph as Subgraph`).

Reply:
Also, this possible issue wouldn't be spotted on a non-ipfs deployment, since the issue would be that the changes within that function (if any) wouldn't be actually stored in the entity after the function execution.

Reply:
I'm pretty sure this works but I'd be happy to return the subgraph entity again. I'll make that change.

Reply:
The one thing i am unsure of is :

- if you pass the entity to `function x()`, `save()` it  in there, and don't return it, and continue forward in the original `function y()`, then the entity in `y()` will not have the updated fields. If you then save again, you overwrite the changes that were made in `x()`. 

I _think_ that might happen. But either way, Jannis' fix means we don't have to worry if this is exactly how it works